### DISCUSSION TITLE
[Q&A] Including Rustdesk password

### DISCUSSION THREAD

**Discussion by dbeniamine** [glpi-project/glpi-agent#749] [Q&A] [answered]
### Documentation & bug reporting acknowledgment

Yes, I read it

### Describe your problem

Hello,

I need to make sure that a hotliner can get from GLPI the id and password for a given computer.

So I tried to:

1. Set a random password in Rustdesk 
2. Use the `additional-content` parameter in glpi-agent to send the password to GLPI

My additional-content file looks like:

```xml
<?xml version="1.0" encoding="UTF-8" ?>
<REQUEST>
<CONTENT>
    <REMOTE_MGMT>
        <ID>myid:mypassword</ID>
        <TYPE>rustdesk</TYPE>
    </REMOTE_MGMT>
</CONTENT>
</REQUEST>
```

This is only a test for now, of course myid is the id and mypassword the actual password.

But when I run `glpi-agent --debug` I got:

```
[error] [http client] proxy error: 400 Bad Request, JSON does not validate. Violations:
Additional properties not allowed: REMOTE_MGMT at #->properties:content
[error] [http client] E8AAB430: server error: JSON does not validate. Violations:
Additional properties not allowed: REMOTE_MGMT at #->properties:content
```

I am not sure what I am doing wrong here, does someone have an idea ?

## technical information 

+ Client
    + Linux mint 20.3
    + Glpi-agent 1.10 (perl installer)
+ Server
    + Debian 12
    + GLPI 10.0.16 (Docker)

**Comment by g-bougard** [comment, marked as answer]
Hi @dbeniamine 

with GLPI 10.0.16 as server backend, additional-content should be a json file.

**Reply by dbeniamine** [reply]
Thanks for your help, it was indeed that.

Just in case someone is trying to do the same thing, here is a small script that gives glpi the right information to create a `rustdesk://` URI including id and password i.e one click connection :

```bash
# Generate a long random password and set it, rustdesk **must** be running
password=$(pwgen -1 64)
rustdesk --password $password
rustdeskjsonpath="/etc/glpi-agent/rustdesk.json"
rustdeskid=$(rustdesk --get-id)
echo "additional-content = $rustdeskjsonpath" >> /etc/glpi-agent/conf.d/00-install.cfg
cat <<EOF > $rustdeskjsonpath
{
  "content": {
    "remote_mgmt": [
      {
        "id": "$rustdeskid/?password=$password",
        "type": "rustdesk"
      }
    ]
  }
}
EOF
```
